extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleSetTableOwner.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.plugin.spark.authz.ranger
+
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.analysis.{PersistedView, ViewType}
+import org.apache.spark.sql.catalyst.catalog.CatalogTable
+import org.apache.spark.sql.catalyst.plans.logical.{Command, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.command.RunnableCommand
+
+import org.apache.kyuubi.plugin.spark.authz.util.AuthZUtils._
+import org.apache.kyuubi.plugin.spark.authz.util.WithInternalChildren
+
+class RuleSetTableOwner(spark: SparkSession) extends Rule[LogicalPlan] {
+  import RuleSetTableOwner._
+
+  private val setOwnerEnabled =
+    spark.conf.get(SET_OWNER_ENABLED_KEY, SET_OWNER_ENABLED_DEFAULT).toLowerCase == "true"
+  private val authzUser = getAuthzUgi(spark.sparkContext).getShortUserName
+  private val catalog = spark.sessionState.catalog
+
+  private val commandToTableField = Map(
+    "CreateDataSourceTableCommand" -> "table",
+    "CreateDataSourceTableAsSelectCommand" -> "table",
+    "CreateTableCommand" -> "table",
+    "CreateHiveTableAsSelectCommand" -> "tableDesc",
+    "OptimizedCreateHiveTableAsSelectCommand" -> "tableDesc")
+
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    if (!setOwnerEnabled) {
+      return plan
+    }
+
+    plan match {
+      case c: Command if commandToTableField.contains(c.nodeName) =>
+        setTableOwner(plan, commandToTableField(c.nodeName))
+        plan
+      case r: RunnableCommand if r.nodeName == "CreateTableLikeCommand" =>
```

Comment:
one more command: AlterTableRenameCommand

Reply:
We should not change table owner after we rename it. 

Take Linux filesystem behavior as a reference: A file's owner is not changed after rename.

Reply:
if not change, after renamed the new table owner will be the HiveClientImpl#userName but not the session user?

Reply:
Checked that new table's owner would be the same as source table's owner.